sensor_msgs/include/sensor_msgs/point_cloud_msg_wrapper.hpp

```diff
@@ -230,17 +231,26 @@ class PointCloudMsgWrapper
   COMPILE_IF_MUTABLE(CloudMsgT, void) push_back(const PointT & point)
   {
     PointT point_copy{point};
-    extend_data_by(m_cloud_ref.point_step);
-    m_cloud_ref.row_step += m_cloud_ref.point_step;
+    extend_data_by(sizeof(PointT));
+    m_cloud_ref.row_step += sizeof(PointT);
     m_cloud_ref.width++;
     this->operator[](m_cloud_ref.width - 1U) = point_copy;
   }
 
+  /// @brief      Push a new point into the message.
+  COMPILE_IF_MUTABLE(CloudMsgT, void) push_back_2(const PointT & point)
+  {
+    m_cloud_ref.row_step += sizeof(PointT);
+    m_cloud_ref.width++;
+    const auto iter_bytes = reinterpret_cast<const unsigned char *>(&point);
+    m_cloud_ref.data.insert(m_cloud_ref.data.end(), iter_bytes, iter_bytes + sizeof(PointT));
+  }
+
   /// @brief      Push a new point into the message.
   COMPILE_IF_MUTABLE(CloudMsgT, void) push_back(PointT && point)
   {
-    extend_data_by(m_cloud_ref.point_step);
-    m_cloud_ref.row_step += m_cloud_ref.point_step;
+    extend_data_by(sizeof(PointT));
```

Comment:
I don't think that this changes anything really. At execution time it is approximately the same infinitesimally small time to lookup a value of a compile-time constant and a value stored in a variable. I don't think we should optimize for that unless we really can measure the impact, which I don't believe we can in this case as it will be too small. That being said, I don't feel strong about it. To me adding a point step is more readable, but adding a size of the point is also fine. So I could merge this into my PR if you feel strongly about it.

Reply:
Keeping things local and not accessing member variables makes things more readable in my opinion. Like `extend data by sizeof PointT` looks very clear to me. Also the fact that it being a constant value in runtime just means it will not be slower than former. But even it had same performance, preferring local values to more global ones is better practice.

Reply:
I buy that. Let's keep this change in and I will merge it once we converge on the other things and clean things up a bit :+1: